<div class="alert alert-block alert-info">
    
<h1 style="color:Blue;"> <center> <ins> <b> 
ESTATÍSTICA APLICADA 
</b> </ins> </center> </h1>
    
<h3 style="color:Blue;"> <center> <b> 
Exercícios 3b
</b></center> </h3>
    
</div>

Nessa atividade, iremos explorar o conjunto de dados `titanic.csv`, que contém informações sobre os passageiros do navio Titanic, que nafraugou no início do século passado. Esse é um conjunto de dados bem famoso, bastante comum em treinamentos e práticas relacionadas a análise de dados e aprendizagem de máquinas.

O conjunto de dados pode ser encontrado nesse [link](https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv).

Para que se possa trabalhar de forma mais consciente, eis um dicionário dos dados coletados nesse conjunto:

* __PassengerId__: Número de identificação do passageiro
* __Survived__: Informa se o passageiro sobreviveu ao desastre
    * 0 = Não
    * 1 = Sim
* __Pclass__: Classe do bilhete
    * 1 = 1ª Classe
    * 2 = 2ª Classe
    * 3 = 3ª Classe
* __Name__: Nome do passageiro
* __Sex__: Sexo do passageiro
* __Age__: Idade do passageiro
* __SibSp__: Quantidade de cônjuges e irmãos a bordo
* __Parch__: Quantidade de pais e filhos a bordo
* __Ticket__: Número da passagem
* __Fare__: Preço da Passagem
* __Cabin__: Número da cabine do passageiro
* __Embarked__: Porto no qual o passageiro embarcou
    * C = Cherbourg
    * Q = Queenstown
    * S = Southampton

Considerando esse conjunto de dados, realize as seguintes atividades:

<div class="alert alert-block alert-info" style="color:Blue;">
1. Carregue o conjunto de dados em um dataframe
</div>

In [92]:
import pandas as pd
import numpy as np
df = pd.read_csv('https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv')
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

<div class="alert alert-block alert-info" style="color:Blue;">
2. Quantas linhas e quantas colunas tem o conjunto de dados?
</div>

In [78]:
df.shape

(891, 12)

891 linhas e 12 colunas

<div class="alert alert-block alert-info" style="color:Blue;">
3. Liste todas as colunas do conjunto de dados.
</div>

In [79]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

<div class="alert alert-block alert-info" style="color:Blue;">
4. Delete as colunas PassengerID e Ticket do conjunto de dados.
</div>

In [80]:
df.drop(['PassengerId','Ticket'], axis=1, inplace=True)
df

Survived  Pclass                                               Name  \
0           0       3                            Braund, Mr. Owen Harris   
1           1       1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2           1       3                             Heikkinen, Miss. Laina   
3           1       1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4           0       3                           Allen, Mr. William Henry   
..        ...     ...                                                ...   
886         0       2                              Montvila, Rev. Juozas   
887         1       1                       Graham, Miss. Margaret Edith   
888         0       3           Johnston, Miss. Catherine Helen "Carrie"   
889         1       1                              Behr, Mr. Karl Howell   
890         0       3                                Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch     Fare Cabin Embarked  
0      male  22.0      1      0   7.2500   NaN        S  
1    female  38.0      1      0  71.2833   C85        C  
2    female  26.0      0      0   7.9250   NaN        S  
3    female  35.0      1      0  53.1000  C123        S  
4      male  35.0      0      0   8.0500   NaN        S  
..      ...   ...    ...    ...      ...   ...      ...  
886    male  27.0      0      0  13.0000   NaN        S  
887  female  19.0      0      0  30.0000   B42        S  
888  female   NaN      1      2  23.4500   NaN        S  
889    male  26.0      0      0  30.0000  C148        C  
890    male  32.0      0      0   7.7500   NaN        Q  

[891 rows x 10 columns]

<div class="alert alert-block alert-info" style="color:Blue;">
5. Selecione as colunas Pclass, Age e Sex, e forme um novo dataframe com essa seleção.
</div>

In [81]:
new_df = df[['Pclass','Age','Sex']]  
new_df

Pclass   Age     Sex
0         3  22.0    male
1         1  38.0  female
2         3  26.0  female
3         1  35.0  female
4         3  35.0    male
..      ...   ...     ...
886       2  27.0    male
887       1  19.0  female
888       3   NaN  female
889       1  26.0    male
890       3  32.0    male

[891 rows x 3 columns]

<div class="alert alert-block alert-info" style="color:Blue;">
6. Exclua, do novo dataframe, todas as linhas que possuam algum valor faltoso.  
</div>

In [82]:
new_df.dropna(inplace=True)
new_df

<ipython-input-82-e045ade23b00>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  new_df.dropna(inplace=True)


Pclass   Age     Sex
0         3  22.0    male
1         1  38.0  female
2         3  26.0  female
3         1  35.0  female
4         3  35.0    male
..      ...   ...     ...
885       3  39.0  female
886       2  27.0    male
887       1  19.0  female
889       1  26.0    male
890       3  32.0    male

[714 rows x 3 columns]

<div class="alert alert-block alert-info" style="color:Blue;">
7. Faça uma filtragem que considere somente passageiros com idade igual ou superior a 35 anos.
</div>

In [83]:
id35 = df.loc[df['Age']>=35]
id35

Survived  Pclass                                               Name  \
1           1       1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
3           1       1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4           0       3                           Allen, Mr. William Henry   
6           0       1                            McCarthy, Mr. Timothy J   
11          1       1                           Bonnell, Miss. Elizabeth   
..        ...     ...                                                ...   
865         1       2                           Bystrom, Mrs. (Karolina)   
871         1       1   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)   
873         0       3                        Vander Cruyssen, Mr. Victor   
879         1       1      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)   
885         0       3               Rice, Mrs. William (Margaret Norton)   

        Sex   Age  SibSp  Parch     Fare Cabin Embarked  
1    female  38.0      1      0  71.2833   C85        C  
3    female  35.0      1      0  53.1000  C123        S  
4      male  35.0      0      0   8.0500   NaN        S  
6      male  54.0      0      0  51.8625   E46        S  
11   female  58.0      0      0  26.5500  C103        S  
..      ...   ...    ...    ...      ...   ...      ...  
865  female  42.0      0      0  13.0000   NaN        S  
871  female  47.0      1      1  52.5542   D35        S  
873    male  47.0      0      0   9.0000   NaN        S  
879  female  56.0      0      1  83.1583   C50        C  
885  female  39.0      0      5  29.1250   NaN        Q  

[235 rows x 10 columns]

<div class="alert alert-block alert-info" style="color:Blue;">
8. Quantos passageiros da terceira classe ao menos 20 anos?
</div>

In [93]:
df1 = df.loc[df['Pclass']==3]
id20 = df1.loc[df['Age']>=20]
print("A quantidade de passageiros da terceira classe com ao menos 20 anos é: ",len(id20.index))

A quantidade de passageiros da terceira classe com ao menos 20 anos é:  247


<div class="alert alert-block alert-info" style="color:Blue;">
9. Quantos passegeiros possuem idade igual ou superior a 50 anos, viajaram na primeira ou na segunda classe e eram do sexo masculino?
</div>

In [94]:
df = df.loc[df['Pclass']<=2]
id50 = df.loc[df['Age']>=50]
masc = id50.loc[df['Sex']=='male']
len(masc.index)

42